mne/connectivity/cfc.py

```diff
@@ -0,0 +1,413 @@
+    as described in [1].
+
+    Parameters
+    ----------
+    data : array, shape (n_times,)
+        Signal time series.
+    sfreq : float
+        Sampling frequency.
+    freq_phase : float
+        The phase modulating frequency in Hz. (eg. 10.) This is the low
+        frequency wave you expect to modify high-frequency amplitude.
+    freqs_amplitude : array, shape (n_freqs,)
+        Frequencies of interest for amplitude modulation. Generally
+        this is a range of frequencies.
+    n_cycles : float
+        Number of cycles used to compute Morlet wavelets.
+    tmin : float, (negative)
+        Start time (s) before phase peak.
+    tmax : float, (positive)
+        End time (s) after phase peak.
+    n_jobs : int
+        Number of jobs to run in parallel. Default 1.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    times : array, shape (n_times,)
+        Time points bef/after a phase peak for signal plotting.
+    raw_spec_amplitude : ndarray (n_freqs, n_times)
+        Time-frequency amplitude traces for amplitude modulated frequencies.
+    evoked_spec_amplitude : ndarray, shape (n_freqs, n_times_epoch)
+        The average evoked spectral amplitude before/after phase peaks. This
+        will have a peri-peak time window where tmin < times < tmax.
+    evoked_raw : array, shape (n_times_epoch,)
+        Average (or evoked) raw signal before/after phase peaks. This
+        will have a peri-peak time window where tmin < times < tmax.
+    trigger_inds : array (trigger_inds,)
+        Indices of maxima in the phase signal.
+
+    References
+    ----------
+    [1] R., Canolty, E., Edwards, SS., Dalal, M., Soltani, SS., Nagarajan,
+        HE., Kirsch, M., Berger, N., Barbaro, R., Knight. High gamma power is
+        phase-locked to theta oscillations in human neocortex. Science, 2006.
+    """
+
+    if data.ndim != 1:
+        raise ValueError('Dimensions of data incorrect. Please use 1d array.')
+
+    Ws = morlet(sfreq, [freq_phase], n_cycles=n_cycles,
+                sigma=None, zero_mean=True)
+    data = data[np.newaxis, :]
+    x_low = cwt(data, Ws, use_fft=False).ravel()
+
+    # Calculate phase of low frequency and find phase peaks for time-locking
+    phases = np.angle(x_low)
+    trigger_inds, _ = peak_finder(phases)
+
+    n_times = data.shape[1]
+    n_freqs = len(freqs_amplitude)
+
+    # obtain min and max indices around phase peaks
+    if tmin > tmax:
+        raise ValueError('tmin must be less than tmax')
+    offset_min = int(round(tmin * sfreq))
+    offset_max = int(round(tmax * sfreq))
+    times = np.arange(offset_min, offset_max, dtype=np.float64) / sfreq
+
+    # makes sure all triggers have enough time bef/after phase peak
+    trigger_inds = trigger_inds[trigger_inds > np.abs(offset_min)]
+    trigger_inds = trigger_inds[trigger_inds < n_times - np.abs(offset_max)]
```

Comment:
why do you use `np.abs` here and not in [line 145](https://github.com/mne-tools/mne-python/pull/2306/files#diff-45dbff1e2ac15094119d5304fb244784R145) ?